niwidgets/streamlines.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import print_function
+import nibabel as nib
+import numpy as np
+from ipywidgets import interact, fixed, widgets
+import os
+import ipyvolume as ipv
+
+
+def length(x):
+    """Returns the sum of euclidean distances between neighboring points"""
+    return np.sum(np.sqrt(np.sum((x[:-1, :] - x[1:, :]) *
+                                 (x[:-1, :] - x[1:, :]),
+                                 axis=1)))
+
+
+def color(x):
+    """Returns an approximation for color of line based on its endpoints"""
+    dirvec = (x[0, :] - x[-1, :])
+    return (dirvec/(np.sqrt(np.sum(dirvec*dirvec, axis=-1)))).dot(np.eye(3))
```

Comment:
Does this do the same thing as this? 

https://github.com/nipy/dipy/blob/master/dipy/viz/colormap.py#L211-L240

I'm not sure it does.

Reply:
given this line, https://github.com/nipy/dipy/blob/master/dipy/viz/colormap.py#L257, i think they are doing exactly the same thing. line 19 can be simplified because i'm not really using anything other than rgb extremes. but instead of eye, one could use three different colors in the colorspace, and a non-linearity on the normalized direction vector.